src/app/shared/services/paginator.service.ts

```diff
@@ -0,0 +1,29 @@
+import {Injectable} from '@angular/core';
+import {GridApi} from 'ag-grid-community';
+import {BehaviorSubject} from 'rxjs';
+import {distinctUntilChanged} from 'rxjs/operators';
+
+export interface IPaginationInfo {
+    pageSize: number;
+    totalItems: number;
+    gridApi: GridApi;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class PaginatorService {
+
+    paginationReady: BehaviorSubject<IPaginationInfo> = new BehaviorSubject({
+        pageSize: 0,
+        totalItems: 0,
```

Comment:
I don't see any reason why we need to carry properties `pageSize` and `totalItem` when they can already be obtained from `gridApi`

Reply:
Earlier, as I was not using the `pageSize` and `rowCount` from `gridApi`, I had to pass these two values explicitly.

 But now, I tried getting the values (not carried) from the api and it's working.

 However, when I change the screen size to test for different screen sizes, `autoPageSize` does not seem to work and it always shows the same `pageSize`.